etna/transforms/encoders/segment_encoder.py

```diff
@@ -44,12 +44,31 @@ def transform(self, df: pd.DataFrame) -> pd.DataFrame:
         -------
         :
             result dataframe
+
+        Raises
+        ------
+        ValueError:
+            If transform isn't fitted.
+        ValueError:
+            If there are segments that weren't present during training.
         """
-        encoded_matrix = self._le.transform(self._le.classes_)
-        encoded_matrix = encoded_matrix.reshape(len(self._le.classes_), -1).repeat(len(df), axis=1).T
+        segments = df.columns.get_level_values("segment").unique().tolist()
+
+        try:
+            new_segments = set(segments) - set(self._le.classes_)
+        except AttributeError:
+            raise ValueError("The transform isn't fitted!")
+
+        if len(new_segments) > 0:
+            raise ValueError(
+                f"This transform can't process segments that weren't present on train data: {new_segments}"
+            )
+
+        encoded_matrix = self._le.transform(segments)
+        encoded_matrix = encoded_matrix.reshape(len(segments), -1).repeat(len(df), axis=1).T
```

Comment:
If I understood correctly, we can do here: `np.tile(encoded_matrix, (len(df), 1))`

Reply:
I will try this. Initially, my intent was to make the smallest change.